File: GeeksForGeeks/Aggressive Cows.cpp
class Solution {
  public:
    bool check(vector<int> &stalls, int k, int min_dist){
        int cows_placed = 1, last_idx = 0;
        for(int i = 1; i < stalls.size(); i++){
            if(abs(stalls[i] - stalls[last_idx]) >= min_dist){
                cows_placed++;
                last_idx = i;
            }
        }
        return cows_placed >= k ? true : false;
    }
    
    int aggressiveCows(vector<int> &stalls, int k) {
        sort(stalls.begin(), stalls.end());
        int low = 0, high = stalls.back() - stalls.front(), ans = 0;
        while(low <= high){
            int mid = low + (high - low) / 2;
            if(check(stalls, k, mid)){
                ans = mid;
                low = mid + 1;
            }
            else    high = mid - 1;
        }
        return ans;
    }
};

File: GeeksForGeeks/Aggressive Cows.py
class Solution:
    def check(self, stalls, min_dist, cows):
        count = 1
        last_placed = 0
        for idx in range(1, len(stalls)):
            if(stalls[idx] - stalls[last_placed] >= min_dist):
                last_placed = idx
                count += 1
        if(count >= cows):
            return True
        else:
            return False
            
    def aggressiveCows(self, stalls, k):
        stalls.sort()
        low = stalls[1] - stalls[0]
        high = stalls[-1] - stalls[0]
        for idx in range(1, len(stalls)):
            low = min(low, stalls[idx] - stalls[idx - 1])
            high = max(high, stalls[idx] - stalls[idx - 1])
            
        while(low <= high):
            mid = low + (high - low) // 2
            if(self.check(stalls, mid, k)):
                answer = mid
                low = mid + 1
            else:
                high = mid - 1
        return answer